src/render/utils/flat-tree.ts

```diff
@@ -0,0 +1,35 @@
+import { addUniqueItem, removeItem } from "../../utils/array"
+import { sortByDepth } from "./sort-by-depth"
+
+export interface DepthChild {
+    depth: number
+}
+
+export class FlatTree {
+    children: DepthChild[] = []
+    numChildren = 0
```

Comment:
It does not look like this is used anywhere else then inside the `forEach` I think it should just move to a local const there